src/lib/frontend/typechecker.ml

```diff
@@ -259,18 +259,29 @@ module Env = struct
     | [ x; y ] -> f x y
     | _ -> assert false
 
-  let add_builtin_enum = function
-    | Ty.Tsum (_, cstrs) as ty ->
-      let enum m h =
-        let s = Hstring.view h in
-        MString.add s (`Term (
-            Symbols.Op (Constr h),
-            { args = []; result = ty },
-            Other
-          )) m
-      in
-      fun m -> List.fold_left enum m cstrs
-    | _ -> assert false
+  let add_fpa_enum map =
+    let ty = Fpa_rounding.fpa_rounding_mode in
+    match ty with
+    | Ty.Tsum (_, cstrs) ->
+      List.fold_left
+        (fun m c ->
+           match Fpa_rounding.translate_smt_rounding_mode c with
+           | None ->
+             (* The constructors of the type are expected to be AE rounding
+                modes. *)
+             assert false
+           | Some hs ->
+             MString.add (Hstring.view hs) (`Term (
+                 Symbols.Op (Constr c),
+                 { args = []; result = ty },
+                 Other
+               ))
+               m
+        )
+        map
+        cstrs
+    | _ -> (* Fpa_rounding.fpa_rounding_mode is a sum type. *)
+      assert false
```

Comment:
IIRC there was a more generic version of this function, which I specialized because it was used only once and I did not expect we would add new enums to the legacy syntax. I can revert it, but I already changed it to simplify the reading.

Reply:
The generic version did not handle renaming so you were right to specialize it. This is a nit, feel free to ignore it :)